nixos/modules/services/web-apps/spliit.nix

```diff
@@ -0,0 +1,245 @@
+        NEXT_PUBLIC_ENABLE_RECEIPT_EXTRACT = true;
+      };
+    };
+
+    database = {
+      createLocally = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Create the PostgreSQL database and database user locally.
+        '';
+      };
+      hostname = mkOption {
+        type = types.str;
+        default = "/run/postgresql";
+        description = "Database hostname.";
+      };
+      port = mkOption {
+        type = types.port;
+        default = 5432;
+        description = "Database port.";
+      };
+      name = mkOption {
+        type = types.str;
+        default = "spliit";
+        description = "Database name.";
+      };
+      user = mkOption {
+        type = types.str;
+        default = "spliit";
+        description = "Database user.";
+      };
+      passwordFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/run/keys/spliit-dbpassword";
+        description = ''
+          A file containing the password corresponding to
+          [](#opt-services.spliit.database.user).
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.spliit = {
+      description = "Spliit bill-splitting web application";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ] ++ lib.optional cfg.database.createLocally "postgresql.service";
+      requires = lib.mkIf cfg.database.createLocally [ "postgresql.service" ];
+
+      environment = lib.mkMerge [
+        (lib.mkIf cfg.database.createLocally {
+          POSTGRES_PRISMA_URL = "postgresql://${cfg.database.user}@${cfg.database.name}/${cfg.database.name}?host=/var/run/postgresql/";
+          POSTGRES_URL_NON_POOLING = "postgresql://${cfg.database.user}@${cfg.database.name}/${cfg.database.name}?host=/var/run/postgresql/";
+        })
+        {
+          PORT = toString cfg.port;
+        }
+        cfg.settings
+      ];
+
+      serviceConfig = {
+        Type = "simple";
+
+        StateDirectory = "spliit";
+        ExecStart =
+          if (cfg.database.passwordFile != null) then
+            lib.getExe (
+              pkgs.writeShellScriptBin "spliit-passwordFile" ''
+                set -eu
+                DATABASE_PASSWORD="$(cat $CREDENTIALS_DIRECTORY/dbpasswordfile)"
+
+                POSTGRES_PRISMA_URL="postgresql://${cfg.database.user}:$DATABASE_PASSWORD@${cfg.database.hostname}:${toString cfg.database.port}/${cfg.database.name}";
+
+                export POSTGRES_PRISMA_URL="$POSTGRES_PRISMA_URL"
+                export POSTGRES_URL_NON_POOLING="$POSTGRES_PRISMA_URL"
+
+                exec ${lib.getExe cfg.package}
+              ''
+            )
+          else
+            lib.getExe cfg.package;
+        Restart = "always";
+        User = cfg.user;
+        Group = cfg.group;
+        LoadCredential =
+          lib.optional (cfg.secretFile != null) "env-secrets:${cfg.secretFile}"
```

Comment:
Are you sure this works? I can't find upstream code that uses `env-secrets`. Isn't `environmentFile` enough?